### PR TITLE
Expose span baggage API

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- Expose `SentrySpanContext` and `SentryBaggage` classes
- Add `baggageHttpHeader` and `traceContext ` on `SentrySpan` protocol

## :bulb: Motivation and Context

Closes [#3519](https://github.com/getsentry/sentry-cocoa/issues/3519)

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps

Exposing `traceContext` in `SentrySpanProtocol` makes it optional, also for `SentryTracer` implementations. Could this be an issue?

### CODE REVIEW

File: Sources/Sentry/SentrySpan.m [hunk 1]
```diff
@@ -297,6 +299,23 @@ - (SentryTraceHeader *)toTraceHeader
...
+        return nil;
+    }
+    return [[self.tracer.traceContext toBaggage] toHTTPHeaderWithOriginalBaggage:nil];
+}
+
+- (nullable SentryTraceContext *)traceContext
+{
+    if (nil == self.tracer) {
+        return nil;
+    }
```
Comment (brustolin) [inline Sources/Sentry/SentrySpan.m:315]: l: This nil check is unnecessary.

```suggestion
```

**Review by brustolin — APPROVED** [review]
LGTM

**Comment by denrase** [comment]
@philipphofmann Need to take another look if the failing CI is just flaky again or if it has to do with this PR.

**Comment by philipphofmann** [comment]
> @philipphofmann Need to take another look if the failing CI is just flaky again or if it has to do with this PR.

@denrase, please merge main back into this branch. I fixed many flaky tests last week. That could fix your problems.

**Comment by denrase** [comment]
@philipphofmann The files were not properly prepared to be public, everything should be fixed now and we can merge.

**Pull request merged**